### PR TITLE
Server: add user-scoped external identity linking API

### PR DESCRIPTION
**Pull request by hughhan1**
## Summary
Adds a host-owned API for linking external identities to the current Gestalt user through the configured `AuthorizationProvider`, using a managed generic `external_identity` resource model.

## Interfaces
- `GET /api/v1/users/me/external-identities`
- `POST /api/v1/users/me/external-identities`
- `DELETE /api/v1/users/me/external-identities/{linkID}`

`POST` request body:
```json
{
  "linkToken": "..."
}
```

`GET` / `POST` response item:
```json
{
  "id": "<opaque-link-id>",
  "externalIdentity": {
    "type": "slack_identity",
    "id": "team:T123:user:U456"
  }
}
```

## Authorization Model
This change extends the managed provider-backed authorization model with:
- resource type: `external_identity`
- relation/action: `assume`
- subject type: `user`

External identity links are persisted as provider-managed relationships and survive `ProviderBackedAuthorizer.ReloadAuthorizationState()`.

## Examples
Link a Slack identity for the current user:
```http
POST /api/v1/users/me/external-identities
Content-Type: application/json

{"linkToken":"<host-issued-token-bound-to-user:user-123>"}
```

List linked external identities:
```http
GET /api/v1/users/me/external-identities
```

Remove a link:
```http
DELETE /api/v1/users/me/external-identities/<opaque-link-id>
```

## Tests
Extended the existing authenticated HTTP server lifecycle coverage with a functional external identity flow that verifies:
- link
- list
- user-bound token enforcement
- conflict on an already-linked identity
- survival across provider-backed authorization reload
- unlink

## Notes
The remaining concurrency risk is backend-dependent: `AuthorizationProvider` does not currently expose a compare-and-swap or uniqueness primitive for cross-request exclusive claims.

### CODE REVIEW

File: gestaltd/internal/server/handlers_external_identities.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+	subjectID := principal.UserSubjectID(userID)
+	for _, rel := range relationships {
+		if rel == nil || rel.GetSubject() == nil {
+			continue
+		}
+		if strings.TrimSpace(rel.GetSubject().GetType()) == "user" && strings.TrimSpace(rel.GetSubject().GetId()) == subjectID {
+			return http.StatusOK, nil
+		}
+		return 0, errExternalIdentityAlreadyLinked
+	}
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply gestaltd/internal/server/handlers_external_identities.go:253]: Fixed on `da804a8c2`. `linkCurrentUserExternalIdentity` now scans the full relationship set, prefers an existing link for the current user, and only returns conflict when another user is linked and the current user is not. I also extended the existing functional server flow test to cover the mixed-order duplicate case.

File: gestaltd/internal/server/handlers_external_identities.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+	subjectID = strings.TrimSpace(subjectID)
+	if principal.UserIDFromSubjectID(subjectID) == "" {
+		return "", errors.New("subjectID must use user:<id>")
+	}
+	return encodeExternalIdentityLinkToken(s.encryptor, externalIdentityLinkTokenState{
+		ExternalIdentity: ref,
+		SubjectID:        subjectID,
+		ExpiresAt:        s.now().Add(externalIdentityLinkTokenTTL).Unix(),
+	})
+}
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply gestaltd/internal/server/handlers_external_identities.go]: Removed in `da804a8c2`. After deleting the unused server method, I also dropped the now-dead shared encode helper so `Go Lint & Vet` stops flagging the branch.
